torch/distributed/_tensor/dispatch.py

```diff
@@ -210,6 +210,57 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
             redistribute_with_schema=needs_redistribute,
         )
 
+        from torch.distributed._tensor.random import (
```

Comment:
I know it's probably not ready as you mark it as draft, but I do want to give some early feedback: we should make sure dispatch.py have as few change as possible, and hide all those calculation inside random.py. This means dispatch.py should only check if this op is random op, if it is, call into some function in random.py that do all the handling there. 

We should have clear separation about each module's purpose. dispatch module is not responsible for read/set offset